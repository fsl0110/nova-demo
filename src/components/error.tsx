import React, { FC, Fragment } from "react";
import { SharedProps } from "../typings";

export interface Props extends SharedProps {
  /** Is loading data */
  isLoading: boolean;
  /** ClickHandler for fetching data */
  fetchData: () => void;
}

export const Error: FC<Props> = ({ isLoading, fetchData }) => (
  <div className="error">
    <div className="text-center">
      <h4>An error occured</h4>
      <p>This Chart can't be shown. Click the reload button to try again.</p>
      <button
        aria-label="Reloadbutton for reloading Chartdatas"
        className="error__button"
        onClick={() => fetchData()}
        disabled={isLoading}
      >
        {isLoading ? (
          <Fragment>
            <span>Loading </span>
            <span className="reload-icon spin">&#8635;</span>
          </Fragment>
        ) : (
          <Fragment>
            <span>Reload</span>
            <span className="reload-icon">&#8635;</span>
          </Fragment>
        )}
      </button>
    </div>
  </div>
);

export default Error;
