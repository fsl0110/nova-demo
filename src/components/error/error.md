### Display fetching errors

#### With isLoading=false

```jsx
<div className="food">
  <div className="chart-wrapper">
    <Error isLoading={false} fetchData={() => null} />
  </div>
</div>
```

#### With isLoading=true

```jsx
<div className="chart-wrapper">
  <Error isLoading={true} fetchData={() => null} />
</div>
```
