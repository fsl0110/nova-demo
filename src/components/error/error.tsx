import React, { FC, HTMLAttributes } from "react";
import { Button } from "..";

export interface Props extends HTMLAttributes<HTMLDivElement> {
  /** Is loading data */
  isLoading?: boolean;
  /** ClickHandler for fetching data */
  fetchData: () => void;
}

export const Error: FC<Props> = ({ isLoading, fetchData }) => (
  <div className="error" data-test="Error">
    <div className="text-center">
      <h4>An error occured</h4>
      <p>This Chart can't be shown. Click the reload button to try again.</p>
      <Button
        aria-label="Reloadbutton for reloading Chartdatas"
        className="error__button"
        iconName="refresh"
        iconAlign="after"
        iconSize="m"
        isLoading={isLoading}
        isLoadingSpin={true}
        onClick={() => fetchData()}
        data-test="error-button"
      >
        {isLoading ? "Loading" : "Reload"}
      </Button>
    </div>
  </div>
);

export default Error;
