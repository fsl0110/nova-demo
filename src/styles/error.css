.error {
  overflow: hidden;
  background-color: pink;
  width: 330px;
  border-radius: 4px;
  display: flex;
  margin: 0 auto;
  justify-content: center;
  align-items: center;
  color: white;
  padding: 2rem 1rem;
  height: 350px;
}

.error h4 {
  font-size: 1.5rem;
  margin: 0;
}

.error p {
  text-align: center;
  font-size: 1.3rem;
  line-height: 1.9rem;
}

.error__button {
  margin-top: 1rem;
  border: 4px solid white;
  background-color: inherit;
  color: white;
  font-size: 1.3rem;
  border-radius: 8px;
  padding: 0.3rem 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-weight: bold;
}

.error__button:disabled {
  cursor: progress;
}

.error__button:disabled {
  border: 4px solid rgb(185, 185, 185);
  color: rgb(185, 185, 185);
}
