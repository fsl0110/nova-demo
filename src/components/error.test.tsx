import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Error, Props } from "./error";

const testProps: Props = {
  isLoading: false,
  fetchData: jest.fn()
};

const testCases: Array<[string, Props]> = [
  [
    "renders with isLoading=true ",
    {
      isLoading: true,
      fetchData: jest.fn()
    }
  ],
  [
    "renders with isLoading=fasle",
    {
      isLoading: false,
      fetchData: jest.fn()
    }
  ]
];

describe("<Error>", () => {
  it("renders without crashing", () => {
    shallow(<Error {...testProps} />);
  });

  testCases.forEach(([item, options]) => {
    it(item, () => {
      const mock = shallow(<Error {...options} />);
      expect(toJson(mock)).toMatchSnapshot();
    });
  });
});
