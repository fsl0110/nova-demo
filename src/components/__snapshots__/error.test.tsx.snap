// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error> renders with isLoading=fasle 1`] = `
<div
  className="error"
>
  <div
    className="text-center"
  >
    <h4>
      An error occured
    </h4>
    <p>
      This Chart can't be shown. Click the reload button to try again.
    </p>
    <button
      aria-label="Reloadbutton for reloading Chartdatas"
      className="error__button"
      disabled={false}
      onClick={[Function]}
    >
      <span>
        Reload
      </span>
      <span
        className="reload-icon"
      >
        ↻
      </span>
    </button>
  </div>
</div>
`;

exports[`<Error> renders with isLoading=true  1`] = `
<div
  className="error"
>
  <div
    className="text-center"
  >
    <h4>
      An error occured
    </h4>
    <p>
      This Chart can't be shown. Click the reload button to try again.
    </p>
    <button
      aria-label="Reloadbutton for reloading Chartdatas"
      className="error__button"
      disabled={true}
      onClick={[Function]}
    >
      <span>
        Loading 
      </span>
      <span
        className="reload-icon spin"
      >
        ↻
      </span>
    </button>
  </div>
</div>
`;
